### PR TITLE
Use minitest and put deps in gemspec

### PR DESCRIPTION
**Pull request by cory-stripe**
#### Summary
Use the gemspec for deps instead of mixing with Gemfile. Also use minitest in of test-unit.

#### Motivation
We evidently prefer minitest, so fixing that. Also we had a mix of deps in two places, so I combined them into one.

#### Test plan
Existing tests pass.

r? @stripe/observability

### CODE REVIEW

File: ssf-ruby.gemspec [hunk 1]
```diff
@@ -13,5 +13,8 @@ spec = Gem::Specification.new do |s|
   s.test_files    = `git ls-files -- test/*`.split("\n")
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ['lib']
-  s.add_runtime_dependency 'google-protobuf', ["= 3.3.0"]
+  s.add_runtime_dependency 'google-protobuf', ['= 3.3.0']
+  s.add_runtime_dependency 'minitest', ['= 5.8.4']
```
Comment (andrew-stripe) [inline ssf-ruby.gemspec]: I think `minitest` and `rake` should be `add_development_dependency` not `add_runtime_dependency`

Comment (cory-stripe) [inline ssf-ruby.gemspec]: Ugh, you're quite right. Thank you for spotting!

**Comment by ChimeraCoder** [comment]
Looks like tests are still failing - you need to run `bundle update` again.

**Comment by cory-stripe** [comment]
Re-approving after rebuild

**Pull request merged**